tools/swift-inspect/Tests/swiftInspectTests/swiftInspectTests.swift

```diff
@@ -12,48 +12,93 @@
-        // This is an example of a functional test case.
-        // Use XCTAssert and related functions to verify your tests produce the correct
-        // results.
+  func testNoConcurrencyActivity() throws {
+    let output = try dumpConcurrency()
+    let (tasks, actors, threads) = try split(output: output)
+    XCTAssertEqual(tasks, [""])
+    // FIXME: sometimes we get a bunch of objects misinterpreted as actors in the actor output
+    XCTAssertEqual(threads, ["  no threads with active tasks"])
+  }
 
-        // Some of the APIs that we use below are available in macOS 10.13 and above.
-        guard #available(macOS 10.13, *) else {
-            return
-        }
+  func testBasicOutput() async throws {
+    actor BasicOutputTestActor {}
+    let actor = BasicOutputTestActor()
+    defer { withExtendedLifetime(actor) {} }
 
-        let fooBinary = productsDirectory.appendingPathComponent("swift-inspect")
+    let output = try dumpConcurrency()
+    let (tasks, actors, threads) = try split(output: output)
 
-        let process = Process()
-        process.executableURL = fooBinary
+    XCTAssertGreaterThan(count(ofSubstring: "Task 0x", in: tasks), 0,
+                         "No tasks present in task output: \(tasks.joined(separator: "\n"))")
+    XCTAssertGreaterThan(count(ofSubstring: "BasicOutputTestActor", in: actors), 0,
+                         "No actors present in actor output: \(actors.joined(separator: "\n"))")
+    XCTAssertGreaterThan(count(ofSubstring: "Thread 0x", in: threads), 0,
+                         "No threads present in thread output: \(threads.joined(separator: "\n"))")
+  }
 
-        let pipe = Pipe()
-        process.standardOutput = pipe
+  private func dumpConcurrency() throws -> [String] {
+    let pid = "\(getpid())"
+    let command = try DumpConcurrency.parseAsRoot(["--fork-corpse", pid])
+    let dumper = try checkedCast(command) as DumpConcurrency
+    var output: [String] = []
+    try dumper.run(printer: { output.append($0) })
+    return output
+  }
 
-        try process.run()
-        process.waitUntilExit()
-
-        let data = pipe.fileHandleForReading.readDataToEndOfFile()
-        let output = String(data: data, encoding: .utf8)
+  private func split(output: [String]) throws
+      -> (tasks: ArraySlice<String>, actors: ArraySlice<String>, threads: ArraySlice<String>) {
+    func find(_ str: String) throws -> Int {
+      if let index = output.firstIndex(of: str) {
+        return index
+      }
+      throw TestError.requiredStringNotPresent(str)
+    }
+    let tasksIndex = try find("TASKS")
+    let actorsIndex = try find("ACTORS")
+    let threadsIndex = try find("THREADS")
+    return (output[tasksIndex..<actorsIndex].dropFirst(),
+            output[actorsIndex..<threadsIndex].dropFirst(),
+            output[threadsIndex...].dropFirst())
+  }
 
-        XCTAssertEqual(output, "Hello, world!\n")
+  private func count(ofSubstring: String, in strings: ArraySlice<String>) -> Int {
```

Comment:
Why not extract this into a freestanding function?

Reply:
No particular reason, just put it near where it's used.

Reply:
Making it freestanding is nicer as it ensures that the compiler realizes it can optimize with this function better.